test/e2e/tests/swap-send/swap-send-eth-with-loaded-state.spec.ts

```diff
@@ -0,0 +1,297 @@
+              options: {
+                chainId: 1,
+                hardfork: 'london',
+                loadState:
+                  './test/e2e/seeder/network-states/swap-state/withSwapContracts.json',
+              },
+            },
+          ],
+        },
+        async ({ driver }) => {
+          const swapSendPage = new SwapSendPage(driver);
+          await logInWithBalanceValidation(driver);
+
+          // disable smart transactions
+          const headerNavbar = new HeaderNavbar(driver);
+          await headerNavbar.check_pageIsLoaded();
+          await headerNavbar.openSettingsPage();
+
+          const settingsPage = new SettingsPage(driver);
+          await settingsPage.check_pageIsLoaded();
+          await settingsPage.clickAdvancedTab();
+          const advancedSettingsPage = new AdvancedSettings(driver);
+          await advancedSettingsPage.check_pageIsLoaded();
+          await advancedSettingsPage.toggleSmartTransactions();
+          await settingsPage.closeSettingsPage();
+
+          // START SWAP AND SEND FLOW
+          await openActionMenuAndStartSendFlow(driver);
+
+          await swapSendPage.fillRecipientAddressInput(RECIPIENT_ADDRESS);
+          await swapSendPage.fillAmountInput('1');
+
+          await swapSendPage.verifyMaxButtonClick(
+            ['ETH', 'ETH'],
+            ['24.99945808355143', '24.99945808355143'],
+          );
+
+          await swapSendPage.fillAmountInput('10');
+          await swapSendPage.verifyAssetSymbolsAndAmounts(
+            [NATIVE_TOKEN_SYMBOL, NATIVE_TOKEN_SYMBOL],
+            ['10', '10'],
+          );
+
+          await swapSendPage.fillAmountInput('10');
+
+          const ETH_WETH_TOKEN_INPUTS = [
+            [NATIVE_TOKEN_SYMBOL, 'WETH'],
+            ['10', '10'],
+          ];
+          const ETH_WETH_FIAT_INPUTS = [
+            ['USD', 'USD'],
+            ['1,700.00', '1,701.09'],
+          ];
+
+          await swapSendPage.clickOnAsset('WETH', 'dest');
+          await swapSendPage.verifyAssetSymbolsAndAmounts(
+            ETH_WETH_TOKEN_INPUTS[0],
+            ETH_WETH_TOKEN_INPUTS[1],
+          );
+
+          await swapSendPage.verifySwitchPrimaryCurrency(
+            ETH_WETH_TOKEN_INPUTS,
+            ETH_WETH_FIAT_INPUTS,
+          );
+
+          await swapSendPage.verifyQuoteDisplay(
+            '1 ETH = 1 WETH',
+            '0.0129028 ETH',
+            '≈ $21.93',
+          );
+
+          // TODO assert swap api request payload
+
+          await swapSendPage.submitSwap();
+          await swapSendPage.verifyHistoryEntry(
+            'Send ETH as WETH',
+            'Confirmed',
+            '-10 ETH',
+            '',
+          );
+
+          // click token tab
+
+          const homePage = new HomePage(driver);
+          await homePage.goToTokensTab();
+          await homePage.check_expectedTokenBalanceIsDisplayed('60', 'WETH');
+          await homePage.check_expectedTokenBalanceIsDisplayed(
+            '14.99994',
```

Comment:
the spec is failing as this value is incorrect

```suggestion
            '14.99995',
```

Reply:
I noticed that it is changing... For some reason is always this on my local and something else in the ci... I'll try with mocks

Reply:
mmm that's interesting :thinking: I could reproduce this consistently in my local. Might it be some rounding issue between OS's :thinking:

Reply:
yes, I think so... for now I have removed the check. 
I also tried to implement a function to pass the number of decimals to check, but the only way I was able to do it was with the anti-pattern const text = await asset.getText();

Reply:
@racitores , I am using a mac machine and locally it failed for me when the value is '14.99994' and test passed when the value is '14.99995'. But it passes in the CI strange.
<img width="1392" alt="Screenshot 2025-03-27 at 2 34 55 PM" src="https://github.com/user-attachments/assets/5a5fcb83-6c12-4aaf-9f5e-5f275a87a916" />

Reply:
this still seems to be an issue. It continues to fail for me locally, same as Harika. And it also failed on ci in previous runs, meaning, there's a gap of knowledge there, that we need to close.

I wouldn't merge the PR until that's clarified, otherwise, this can start failing in ci anytime this unknown behaviour surfaces :thinking: 

![Screenshot from 2025-03-28 15-22-35](https://github.com/user-attachments/assets/726d8097-d4e5-41cd-b1e0-dd7c8df07585)

Reply:
if you don't want to block this PR, I would suggest to maybe comment this line,  create a follow-up ticket and reference the ticket number with a comment in the code :thinking: what do you think?

Reply:
https://github.com/MetaMask/metamask-extension/issues/31427